suites/create.js

```diff
@@ -245,6 +245,19 @@ export function runDataIntegrityProofFormatTests({
         reason: 'Expected issuer to error when VC has an undefined term.'
       });
     });
+    it('If the algorithm produces an error, the error MUST be propagated ' +
+      'and SHOULD convey the error type.', async function() {
+      this.test.link = 'https://w3c.github.io/vc-data-integrity/#dfn-input-document:~:text=If%20the%20algorithm%20produces%20an%20error%2C%20the%20error%20MUST%20be%20propagated%20and%20SHOULD%20convey%20the%20error%20type';
+      const noSubject = structuredClone(credential);
+      delete noSubject.credentialSubject;
+      // FIXME this test should look for a propagated error such as
+      // PROOF_GENERATION_ERROR
```

Comment:
Is this fixable before we merge this one?

Reply:
we need to inform the implementers that they need to start returning error objects when issuance fails and verification fails. Currently most implementers do return verification results. Basically we need to make implementers ware that https://w3c-ccg.github.io/vc-api/#error-handling VC-API Error Handling is a thing. I also thought the VC-API group came to consensus at some point about standardized error handling, but it looks like they have not although error codes in the specs do suggest there is a common error object for data integrity at least. So perhaps we should ask implementers for the test suite to return https://w3c.github.io/vc-data-integrity/#processing-errors Problem Details in the future. Basically right now we not have informed implementers to return problem details so asserting on those error codes in data returned on fail seems unfair. I can add a section to the README about this as we really need those problem details for the suite.

Reply:
@BigBlueHat revised README is here: https://github.com/w3c-ccg/data-integrity-test-suite-assertion/pull/122

Those details should probably be in the implementations README to.

Reply:
https://github.com/w3c-ccg/vc-api/pull/418

this is the PR that defines it, ideally there would be a `problemDetails` property in the response and we just need to ensure there's at least 1 item in that array, since there's no instruction on what to look for in that statement.